### PR TITLE
fix: empty string as license key stuck docker compose

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title, we do a license check when bootstrap, we actually get an empty string and try to interpret it as a license key, leading to a check error on compute node. 
We expect it to be NULL and start the server as a free-tier. 

The case does not cause any trouble because the check is introduced in v2.0, which has not been rolled out. 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by tabVersion** [comment]
& does this pr need to be cherry-picked to release-2.0?

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -60,7 +60,6 @@ services:
       ENABLE_TELEMETRY: ${ENABLE_TELEMETRY:-true}
       RW_TELEMETRY_TYPE: ${RW_TELEMETRY_TYPE:-"docker-compose"}
       RW_SECRET_STORE_PRIVATE_KEY_HEX: ${RW_SECRET_STORE_PRIVATE_KEY_HEX:-0123456789abcdef}
-      RW_LICENSE_KEY: ${RW_LICENSE_KEY:-""}
```
Comment (BugenZhao) [inline docker/docker-compose.yml:63]: I think we should still pass the env var? ...but only remove the quotes.

```yaml
RW_LICENSE_KEY: ${RW_LICENSE_KEY:-}
```

Comment (tabVersion) [reply docker/docker-compose.yml:63]: cannot tell the diff here. But after removing the line, it works.

Comment (BugenZhao) [reply docker/docker-compose.yml:63]: Only env vars specified here will inherit from the host.

**Review by BugenZhao — APPROVED** [review]

**Review by huangjw806 — APPROVED** [review]

**Pull request merged**